### ISSUE TITLE
ERROR: Could not load URLWebDriverError: chrome not reachable - docker sitespeedio/sitespeed.io:8.11.0-plus1

### ISSUE THREAD

**Issue by washingon** [sitespeedio/sitespeed.io#2416]
On ubuntu 16.04:    -b firefox works, -b chrome does not
same error with sitespeedio/sitespeed.io:8.11.0

~$ docker run --shm-size=1g --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.11.0-plus1 -b chrome http://www.sitespeed.io/
Google Chrome 73.0.3683.75
Mozilla Firefox 66.0
[2019-04-09 11:40:20] INFO: Versions OS: linux 4.4.0-143-generic nodejs: v10.13.0 sitespeed.io: 8.11.0 browsertime: 4.6.0 coach: 3.3.0
[2019-04-09 11:40:21] INFO: Will run Lighthouse tests after Browsertime has finished
[2019-04-09 11:40:22] INFO: Sending url http://www.sitespeed.io/ to test on Page Speed Insights
[2019-04-09 11:40:22] INFO: Running tests using Chrome - 3 iteration(s)
[2019-04-09 11:40:23] INFO: Testing url http://www.sitespeed.io/ iteration 1
[2019-04-09 11:40:24] INFO: Got http://www.sitespeed.io/ analysed from Google Page Speed Insights
[2019-04-09 11:40:25] ERROR: Could not load URLWebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.4.0-143-generic x86_64)
[2019-04-09 11:40:25] ERROR: UrlLoadError: Failed to load http://www.sitespeed.io/
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-09 11:40:26] INFO: Testing url http://www.sitespeed.io/ iteration 2
[2019-04-09 11:40:27] ERROR: Could not load URLWebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.4.0-143-generic x86_64)
[2019-04-09 11:40:27] ERROR: UrlLoadError: Failed to load http://www.sitespeed.io/
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-09 11:40:28] INFO: Testing url http://www.sitespeed.io/ iteration 3
[2019-04-09 11:40:29] ERROR: Failed to wait WebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.4.0-143-generic x86_64)
[2019-04-09 11:40:29] ERROR: Could not load URLUrlLoadError: Failed to wait
[2019-04-09 11:40:29] ERROR: UrlLoadError: Failed to load http://www.sitespeed.io/
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-09 11:40:29] INFO: http://www.sitespeed.io/ 0
[2019-04-09 11:40:29] INFO: Will collect Lighthouse metrics for http://www.sitespeed.io/
[2019-04-09 11:40:35] INFO: Got Lighthouse metrics
[2019-04-09 11:40:37] INFO: HTML stored in /sitespeed.io/sitespeed-result/www.sitespeed.io/2019-04-09-11-40-20

with DEBUG:
$ docker run --shm-size=1g --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.11.0-plus1 -b chrome http://www.sitespeed.io/ --debug
Google Chrome 73.0.3683.75
Mozilla Firefox 66.0
[2019-04-09 11:44:41] INFO: Versions OS: linux 4.4.0-143-generic nodejs: v10.13.0 sitespeed.io: 8.11.0 browsertime: 4.6.0 coach: 3.3.0
[2019-04-09 11:44:42] INFO: {
  "uuid": "c0b5fff6-e630-48f5-bcd0-d779f7d0993f",
  "type": "sitespeedio.setup",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "queueHandler"
}
[2019-04-09 11:44:42] INFO: Will run Lighthouse tests after Browsertime has finished
[2019-04-09 11:44:42] INFO: {
  "uuid": "2dc42901-a4ac-4af7-8ba6-557a44116bd7",
  "type": "browsertime.setup",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "browsertime"
}
[2019-04-09 11:44:42] INFO: {
  "uuid": "256967ad-5a8f-4553-8bd8-f755c8ee212a",
  "type": "browsertime.config",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "browsertime",
  "data": "{...}"
}
[2019-04-09 11:44:42] INFO: {
  "uuid": "0b66a820-b0ab-492a-a263-9ca24f4ff674",
  "type": "browsertime.config",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "coach",
  "data": "{...}"
}
[2019-04-09 11:44:42] INFO: {
  "uuid": "ce866351-1f9e-493e-875f-857ad2f5be2b",
  "type": "html.pug",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "lighthouse",
  "data": "{...}"
}
[2019-04-09 11:44:42] INFO: {
  "uuid": "85a9c125-c0c5-4621-9644-ba78022727c2",
  "type": "budget.addMessageType",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "lighthouse",
  "data": "{...}"
}
[2019-04-09 11:44:42] INFO: {
  "uuid": "8e7d3dac-295a-4ca5-96f5-74d32371b576",
  "type": "html.pug",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "gpsi",
  "data": "{...}"
}
[2019-04-09 11:44:42] INFO: {
  "uuid": "9653d271-de15-4cb8-b712-87fb790ce292",
  "type": "budget.addMessageType",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "gpsi",
  "data": "{...}"
}
[2019-04-09 11:44:42] INFO: {
  "uuid": "df2101d4-ae2c-4b8b-a31b-c52e506128a1",
  "type": "url",
  "timestamp": "2019-04-09T11:44:42+00:00",
  "source": "url-reader",
  "data": {},
  "url": "http://www.sitespeed.io/",
  "group": "www.sitespeed.io"
}
[2019-04-09 11:44:42] INFO: Sending url http://www.sitespeed.io/ to test on Page Speed Insights
[2019-04-09 11:44:42] INFO: Running tests using Chrome - 3 iteration(s)
[2019-04-09 11:44:43] INFO: Testing url http://www.sitespeed.io/ iteration 1
[2019-04-09 11:44:45] ERROR: Failed to wait WebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.4.0-143-generic x86_64)
[2019-04-09 11:44:45] ERROR: Could not load URLUrlLoadError: Failed to wait
[2019-04-09 11:44:45] ERROR: UrlLoadError: Failed to load http://www.sitespeed.io/
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-09 11:44:45] INFO: Got http://www.sitespeed.io/ analysed from Google Page Speed Insights
[2019-04-09 11:44:45] INFO: {
  "uuid": "00389014-5913-4393-97ea-73faa34f3f9f",
  "type": "gpsi.pageSummary",
  "timestamp": "2019-04-09T11:44:45+00:00",
  "source": "gpsi",
  "data": "{...}",
  "url": "http://www.sitespeed.io/",
  "group": "www.sitespeed.io"
}
[2019-04-09 11:44:45] INFO: Testing url http://www.sitespeed.io/ iteration 2
[2019-04-09 11:44:47] ERROR: Could not load URLWebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.4.0-143-generic x86_64)
[2019-04-09 11:44:47] ERROR: UrlLoadError: Failed to load http://www.sitespeed.io/
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-09 11:44:48] INFO: Testing url http://www.sitespeed.io/ iteration 3
[2019-04-09 11:44:50] ERROR: Failed to wait WebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.4.0-143-generic x86_64)
[2019-04-09 11:44:50] ERROR: Could not load URLUrlLoadError: Failed to wait
[2019-04-09 11:44:50] ERROR: UrlLoadError: Failed to load http://www.sitespeed.io/
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-09 11:44:50] INFO: http://www.sitespeed.io/ 0
[2019-04-09 11:44:50] INFO: {
  "uuid": "bd622596-34df-4d6c-a004-5f56e50eaeef",
  "type": "browsertime.pageSummary",
  "timestamp": "2019-04-09T11:44:50+00:00",
  "source": "browsertime",
  "data": "{...}",
  "url": "http://www.sitespeed.io/",
  "group": "www.sitespeed.io"
}
[2019-04-09 11:44:50] INFO: {
  "uuid": "9fd9c1d5-d4a4-4a5c-9ee9-f49b238392bc",
  "type": "error",
  "timestamp": "2019-04-09T11:44:50+00:00",
  "source": "browsertime",
  "data": "UrlLoadError",
  "url": "http://www.sitespeed.io/"
}
[2019-04-09 11:44:50] INFO: {
  "uuid": "67d00abc-113f-4446-ac5f-7ec1820ecb61",
  "type": "error",
  "timestamp": "2019-04-09T11:44:50+00:00",
  "source": "browsertime",
  "data": "UrlLoadError",
  "url": "http://www.sitespeed.io/"
}
[2019-04-09 11:44:50] INFO: {
  "uuid": "b8f7696f-5f71-4a0f-aeaf-784bbba0dd84",
  "type": "error",
  "timestamp": "2019-04-09T11:44:50+00:00",
  "source": "browsertime",
  "data": "UrlLoadError",
  "url": "http://www.sitespeed.io/"
}
[2019-04-09 11:44:50] INFO: Will collect Lighthouse metrics for http://www.sitespeed.io/
[2019-04-09 11:44:56] INFO: Got Lighthouse metrics
[2019-04-09 11:44:56] INFO: {
  "uuid": "4e78df22-614e-4f85-9340-a89b6ca05414",
  "type": "lighthouse.pageSummary",
  "timestamp": "2019-04-09T11:44:56+00:00",
  "source": "lighthouse",
  "data": "{...}",
  "url": "http://www.sitespeed.io/",
  "group": "www.sitespeed.io"
}
[2019-04-09 11:44:56] INFO: {
  "uuid": "17141496-2f45-4382-9bc2-3a3b381afc0a",
  "type": "lighthouse.report",
  "timestamp": "2019-04-09T11:44:56+00:00",
  "source": "lighthouse",
  "data": "{\n  \"userAgent\": \"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Headless...",
  "url": "http://www.sitespeed.io/",
  "group": "www.sitespeed.io"
}
[2019-04-09 11:44:56] INFO: {
  "uuid": "832517d4-5750-46c1-a0a9-866b91d8db37",
  "type": "sitespeedio.summarize",
  "timestamp": "2019-04-09T11:44:56+00:00",
  "source": "queueHandler"
}
[2019-04-09 11:44:56] INFO: {
  "uuid": "8e2705bf-f260-4430-b48d-4d4fbf9aab8a",
  "type": "aggregateassets.summary",
  "timestamp": "2019-04-09T11:44:56+00:00",
  "source": "assets",
  "data": {},
  "group": "total"
}
[2019-04-09 11:44:56] INFO: {
  "uuid": "840b5b61-c641-48c6-b77e-cf437aba194f",
  "type": "largestassets.summary",
  "timestamp": "2019-04-09T11:44:56+00:00",
  "source": "assets",
  "data": {},
  "group": "total"
}
[2019-04-09 11:44:56] INFO: {
  "uuid": "3c8bafbc-0159-4189-9d84-f411c139b8de",
  "type": "slowestassets.summary",
  "timestamp": "2019-04-09T11:44:56+00:00",
  "source": "assets",
  "data": {},
  "group": "total"
}
[2019-04-09 11:44:56] INFO: {
  "uuid": "b13d44c8-ec64-4519-bf85-81dd74400e5b",
  "type": "domains.summary",
  "timestamp": "2019-04-09T11:44:56+00:00",
  "source": "domains",
  "data": {},
  "group": "total"
}
[2019-04-09 11:44:56] INFO: {
  "uuid": "3f31bb9e-c154-4ef1-90da-6c569a3188f6",
  "type": "sitespeedio.render",
  "timestamp": "2019-04-09T11:44:56+00:00",
  "source": "queueHandler"
}
[2019-04-09 11:44:57] INFO: HTML stored in /sitespeed.io/sitespeed-result/www.sitespeed.io/2019-04-09-11-44-41
[2019-04-09 11:44:57] INFO: {
  "uuid": "d9fbc554-43d3-484b-a421-2f9282045ac3",
  "type": "html.finished",
  "timestamp": "2019-04-09T11:44:57+00:00",
  "source": "html"
}

**Comment by washingon** [comment]
docker -v
Docker version 18.06.1-ce, build e68fc7a

lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.5 LTS
Release:        16.04
Codename:       xenial

**Comment by soulgalore** [comment]
Hi @washingon thanks for the report. It was really long time ago since I saw that error. The containers work for me, so I wonder what it can be. Do you have anything else running on that machine? Thinking something breaks the communication between Chrome/Chromedriver?

Best
Peter

**Comment by washingon** [comment]
thanks @soulgalore  I used exact same command on 2 different ubuntu boxes .. one is 16.04 and the other 18.04.  I wanted to keep the test for chrome as simple as possible.  the 16.04 is the GUI version and 18.04 is ubuntu server.  Its all II had available.   firefox browser works great.

If it helps, 8.7.5 works with -b chrome.  8.8.0 and later all fail.  I tried most of the tagged releases with exact same command from 8.6.5 to 8.11.0 today on the 16.04 ubuntu box just to see where the break might have occurred. -b firefox works great all through. and 8.7.5 -b chrome works.

the ubuntu 16.04 that I did most of the testing on has nginx running on it.

Replicated same error with a brand new AWS EC2 ubuntu 18.04 instance that I just spun up on t2.small for testing.  See output below.  You should be able to replicate issue on new aws ec2 ubuntu instance.

AWS ubuntu 18.04 EC2:
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.2 LTS
Release:        18.04
Codename:       bionic

$ docker -v
Docker version 18.06.1-ce, build e68fc7a

$ sudo docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.11.0 -b chrome https://www.sitespeed.io/ -n 1
Google Chrome 73.0.3683.75
Mozilla Firefox 66.0
[2019-04-09 23:48:44] INFO: Versions OS: linux 4.15.0-1032-aws nodejs: v10.13.0 sitespeed.io: 8.11.0 browsertime: 4.6.0 coach: 3.3.0
[2019-04-09 23:48:45] INFO: Running tests using Chrome - 1 iteration(s)
[2019-04-09 23:48:47] INFO: Testing url https://www.sitespeed.io/ iteration 1
[2019-04-09 23:48:48] ERROR: Could not load URLWebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.15.0-1032-aws x86_64)
[2019-04-09 23:48:48] ERROR: UrlLoadError: Failed to load https://www.sitespeed.io/
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-09 23:48:48] INFO: https://www.sitespeed.io/ 0
[2019-04-09 23:48:50] INFO: HTML stored in /sitespeed.io/sitespeed-result/www.sitespeed.io/2019-04-09-23-48-44

I'm stuck using 8.7.5 because I need chrome and firefox for my testing until this is resolved.  What can I run to help diagnose and resolve the issue?

**Comment by soulgalore** [comment]
Thanks for all the info @washingon !  8.8.0 upgraded to Chrome 73, that's the only thing I could see that could be related.  We run Ubuntu both on AWS and Digital Ocean on the test servers, and there it works so really strange, let me try a fresh box.

**Comment by soulgalore** [comment]
Ah I checked the coming Chromedriver (74):
 
> Resolved issue 2782: ChromeDriver can't connect to Chrome when /dev/shm is not available on Linux [[Pri-2]]

https://chromedriver.storage.googleapis.com/74.0.3729.6/notes.txt

That could be it. Let me upgrade and see that the beta Chromedriver works for 73.

**Comment by soulgalore** [comment]
No that didn't work because nowadays the specific Chromedriver must match a specific Chrome version. Can you re-run and just remove `--shm-size=1g `, you shouldn't need that since 8.0 ? Or setup https://www.cyberciti.biz/tips/what-is-devshm-and-its-practical-usage.html

**Comment by washingon** [comment]
@soulgalore  

on aws ubuntu 18.04 ec2 with sitespeed 8.11.1

sudo docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.11.1 -b chrome https://sitespeed.io -n 1
Unable to find image 'sitespeedio/sitespeed.io:8.11.1' locally
8.11.1: Pulling from sitespeedio/sitespeed.io
Digest: sha256:46cac9e3316e0de9d46a7c02b81b94aa4a089f5bbee45fafa60266ca635c50a5
Status: Downloaded newer image for sitespeedio/sitespeed.io:8.11.1
Google Chrome 73.0.3683.75
Mozilla Firefox 66.0
[2019-04-10 12:03:50] INFO: Versions OS: linux 4.15.0-1032-aws nodejs: v10.13.0 sitespeed.io: 8.11.1 browsertime: 4.6.2 coach: 3.3.0
[2019-04-10 12:03:51] INFO: Running tests using Chrome - 1 iteration(s)
[2019-04-10 12:03:55] INFO: Testing url https://sitespeed.io iteration 1
[2019-04-10 12:03:55] ERROR: Could not load URLWebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.15.0-1032-aws x86_64)
[2019-04-10 12:03:55] ERROR: UrlLoadError: Failed to load https://sitespeed.io
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-10 12:03:55] INFO: https://sitespeed.io 0
[2019-04-10 12:03:57] INFO: HTML stored in /sitespeed.io/sitespeed-result/sitespeed.io/2019-04-10-12-03-50


rerunning 8.11.0
on aws ubuntu 18.04 ec2  with sitespeed 8.11.0

sudo docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.11.0 -b chrome https://sitespeed.io -n 1
Google Chrome 73.0.3683.75
Mozilla Firefox 66.0
[2019-04-10 12:05:45] INFO: Versions OS: linux 4.15.0-1032-aws nodejs: v10.13.0 sitespeed.io: 8.11.0 browsertime: 4.6.0 coach: 3.3.0
[2019-04-10 12:05:45] INFO: Running tests using Chrome - 1 iteration(s)
[2019-04-10 12:05:47] INFO: Testing url https://sitespeed.io iteration 1
[2019-04-10 12:05:48] ERROR: Failed to wait WebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.15.0-1032-aws x86_64)
[2019-04-10 12:05:48] ERROR: Could not load URLUrlLoadError: Failed to wait
[2019-04-10 12:05:48] ERROR: UrlLoadError: Failed to load https://sitespeed.io
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-10 12:05:48] INFO: https://sitespeed.io 0
[2019-04-10 12:05:50] INFO: HTML stored in /sitespeed.io/sitespeed-result/sitespeed.io/2019-04-10-12-05-45

**Comment by soulgalore** [comment]
I would guess then that you are using a too small instance, Chrome needs memory. I just tested on a c4.large with Ubuntu Server 18.04 LTS (HVM), SSD Volume Type, it worked.

**Comment by washingon** [comment]
I just created new AWS ec2 c4.large with Ubuntu Server 18.04 LTS (HVM), SSD Volume Type 64-bit (x86).

sudo snap install docker
sudo docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:8.11.1 -b chrome https://sitespeed.io -n 1

same errors

Status: Downloaded newer image for sitespeedio/sitespeed.io:8.11.1
Google Chrome 73.0.3683.75
Mozilla Firefox 66.0
[2019-04-11 02:47:35] INFO: Versions OS: linux 4.15.0-1032-aws nodejs: v10.13.0 sitespeed.io: 8.11.1 browsertime: 4.6.2 coach: 3.3.0
[2019-04-11 02:47:35] INFO: Running tests using Chrome - 1 iteration(s)
[2019-04-11 02:47:36] INFO: Testing url https://sitespeed.io iteration 1
[2019-04-11 02:47:37] ERROR: Could not load URLWebDriverError: chrome not reachable
  (Session info: chrome=73.0.3683.75)
  (Driver info: chromedriver=73.0.3683.68 (47787ec04b6e38e22703e856e101e840b65afe72),platform=Linux 4.15.0-1032-aws x86_64)
[2019-04-11 02:47:37] ERROR: UrlLoadError: Failed to load https://sitespeed.io
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:223:13)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2019-04-11 02:47:37] INFO: https://sitespeed.io 0
[2019-04-11 02:47:38] INFO: HTML stored in /sitespeed.io/sitespeed-result/sitespeed.io/2019-04-11-02-47-35

What am I doing different?

**Comment by gidztech** [comment]
I'm also getting this error when I run in CI with EC2 Ubuntu instances. My logs look much the same. It has worked previously.

**Comment by soulgalore** [comment]
Hi @washingon thanks for sharing how you did, I could reproduce that way too. Installing Docker with snap also makes it fails for me. Looking at the chromedriver log I see:

```
[1554964809.566][SEVERE]: bind() failed: Cannot assign requested address (99)
[1554964809.566][INFO]: listen on IPv6 failed with error ERR_ADDRESS_INVALID
...
[1554964811.311][DEBUG]: failed to connect to localhost (error -108)
[1554964811.311][DEBUG]: DevTools HTTP Request: http://localhost:38131/json
[1554964811.311][DEBUG]: DevTools HTTP Request failed
[1554964811.311][INFO]: [ac53b5989969fc5eec8b8d95b5eb465d] RESPONSE ExecuteScript ERROR chrome not reachable
  (Session info: chrome=73.0.3683.75)
[1554964811.324][INFO]: [ac53b5989969fc5eec8b8d95b5eb465d] COMMAND Quit {
}
```

However installing Docker as in https://docs.docker.com/install/linux/docker-ce/ubuntu/ then it works for me.

Using snap I get:
`Doccker version 18.06.1-ce, build e68fc7a`

Using raw install I get version:
`Docker version 18.09.5, build e8ff056`

I wonder if it is a version thing or a installation thing? @gidztech can you see what version you are running?

In the Chromedriver log it looks like a ipv6 issue for Docker but I wonder if the [documentation in Docker](https://docs.docker.com/config/daemon/ipv6/ ) is uptodate?

Best
Peter

**Comment by soulgalore** [comment]
There's another user that reported that upgrading to the same version that worked for me, didn't help them. I've built a new image with Chrome 74 and Chromedriver and that works for me when I install `sudo snap install docker`

`sudo docker run --rm -v "$(pwd)":/browsertime sitespeedio/browsertime-autobuild:chrome-74-beta -n 1 https://www.sitespeed.io`

I can upgrade the current containers tonight, but it would be great someone else also could verify that the new Chrome container work.

**Comment by washingon** [comment]
success.  

success - on ubuntu 18.04 t2.small ec2 snap install docker  (same t2.small that was failing with 8.11.1.  no change to docker):
sudo docker run --rm -v "$(pwd)":/browsertime sitespeedio/browsertime-autobuild:chrome-74-beta -n 1 https://www.sitespeed.io
Google Chrome 74.0.3729.75 beta
Mozilla Firefox 66.0
[2019-04-11 12:16:18] INFO: Running tests using Chrome - 1 iteration(s)
[2019-04-11 12:16:19] INFO: Testing url https://www.sitespeed.io iteration 1
[2019-04-11 12:16:35] INFO: https://www.sitespeed.io 12 requests, 141.29 kb, backEndTime: 141ms, firstPaint: 468ms, firstVisualChange: 466ms, DOMContentLoaded: 289ms, Load: 422ms, speedIndex: 466, visualComplete85: 466ms, lastVisualChange: 466ms, rumSpeedIndex: 468
[2019-04-11 12:16:35] INFO: Wrote data to browsertime-results/www.sitespeed.io/2019-04-11T121617+0000

success - ubuntu 160.4 with docker-ce installed per ubuntu instructions (not an aws ec2 instance)
sudo docker run --rm -v "$(pwd)":/browsertime sitespeedio/browsertime-autobuild:chrome-74-beta -n 1 https://www.sitespeed.io
Google Chrome 74.0.3729.75 beta
Mozilla Firefox 66.0
[2019-04-11 12:19:02] INFO: Running tests using Chrome - 1 iteration(s)
[2019-04-11 12:19:03] INFO: Testing url https://www.sitespeed.io iteration 1
[2019-04-11 12:19:18] INFO: https://www.sitespeed.io 12 requests, 141.29 kb, backEndTime: 996ms, firstPaint: 1.16s, firstVisualChange: 1.17s, DOMContentLoaded: 1.14s, Load: 1.17s, speedIndex: 1167, visualComplete85: 1.17s, lastVisualChange: 1.17s, rumSpeedIndex: 1160
[2019-04-11 12:19:18] INFO: Wrote data to browsertime-results/www.sitespeed.io/2019-04-11T121902+0000

success - ubuntu 18.04 (not aws ec2) that failed with prior 8.11.1.
sudo docker run --rm -v "$(pwd)":/browsertime sitespeedio/browsertime-autobuild:chrome-74-beta -n 1 https://www.sitespeed.io
Google Chrome 74.0.3729.75 beta
Mozilla Firefox 66.0
[2019-04-11 12:20:26] INFO: Running tests using Chrome - 1 iteration(s)
[2019-04-11 12:20:27] INFO: Testing url https://www.sitespeed.io iteration 1
[2019-04-11 12:20:38] INFO: https://www.sitespeed.io 12 requests, 141.29 kb, backEndTime: 208ms, firstPaint: 350ms, firstVisualChange: 366ms, DOMContentLoaded: 284ms, Load: 349ms, speedIndex: 366, visualComplete85: 366ms, lastVisualChange: 366ms, rumSpeedIndex: 350
[2019-04-11 12:20:38] INFO: Wrote data to browsertime-results/www.sitespeed.io/2019-04-11T122026+0000

**Comment by soulgalore** [comment]
Thank you @washingon for trying it out, I'll let it run on dashboard,sitespeed.io for a while to make sure it doesn't cause other problems, and will push a new release if everything looks ok later tonight.

**Comment by soulgalore** [comment]
I couldn't see any change in metrics with 74 (good!) but Chrome has added some new event that we pickup as "other" as time spent, so I need to look into that and check what it is. After that I can do a new release where the container includes 74.

**Comment by soulgalore** [comment]
I've pushed 8.12.0 with the latest beta of 74 that hopefully solves this.

**Comment by washingon** [comment]
8.12.0 resolves bug #2416

success - ubuntu 16.04 with docker-ce installed per ubuntu instructions (not an aws ec2 instance)
sudo docker run --rm -v "$(pwd)":/browsertime sitespeedio/sitespeed.io:8.12.0 -n 1 https://www.sitespeed.io

success - ubuntu 18.04 (not aws ec2) that failed previously with 8.11.0 (not an aws ec2 instance)
sudo docker run --rm -v "$(pwd)":/browsertime sitespeedio/sitespeed.io:8.12.0 -n 1 https://www.sitespeed.io

**Comment by gidztech** [comment]
I can also confirm that the problem is resolved from me using `8.12.0-plus1`. It would probably be ideal to have a stable Chrome instead normally, but for now this should be fine. Thanks for that.

**Comment by soulgalore** [comment]
I'm not too worried over running beta, we have been doing that a couple of times already over the years when the eco-system has been broken. I think this is the third time there's been issues between Chromedriver/Chrome and I think we almost run Firefox one year in the old version since 57 broke getting a HAR file. I'll update to stable ASAP it is released.
